File: app/src/main/java/igrek/todotree/domain/stats/StatisticEventType.java
package igrek.todotree.domain.stats;

import java.text.ParseException;

public enum StatisticEventType {
	
	TASK_CREATED("task-created"),
	
	TASK_COMPLETED("task-completed");
	
	private String name;
	
	StatisticEventType(String name) {
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
	
	public static StatisticEventType parse(String name) throws ParseException {
		for (StatisticEventType statisticEventType : values()) {
			if (statisticEventType.name.equals(name))
				return statisticEventType;
		}
		throw new ParseException("unknown statistic event type name: " + name, 0);
	}
}

File: app/src/main/java/igrek/todotree/domain/stats/StatisticEventType.kt
package igrek.todotree.domain.stats

import java.text.ParseException

enum class StatisticEventType(val givenName: String) {
    TASK_CREATED("task-created"), TASK_COMPLETED("task-completed");

    companion object {
        @Throws(ParseException::class)
        fun parse(givenName: String): StatisticEventType {
            for (statisticEventType in values()) {
                if (statisticEventType.givenName == givenName) return statisticEventType
            }
            throw ParseException("unknown statistic event type name: $givenName", 0)
        }
    }

}